### DISCUSSION TITLE
How to send authentication payload?

### DISCUSSION THREAD

**Discussion by paramsiddharth** [socketio/socket.io#4072] [Q&A] [answered]
I am trying to use Postman to "connect" to a Socket.io server.

Using a regular client in NodeJS would look like this:
```js
const io = io(url, {
    auth: {
        token: 123
    }
});
```

How to I pass this `auth` information while making a manual Socket connection? The documentations suggest that the new `auth` property is just `query` renamed, but I am unable to use query parameters in Postman to get it to work.

Yes, WebSocket requests in Postman are in beta. That could very much be the reason why I can't get it to work, but if any of you has tried this before and got it to work, please let me know. Thank you!

**Comment by darrachequesne** [comment, marked as answer]
Hi! You must include the `auth` object in the CONNECT packet:

```
40{"token":"123"}

- 4 for Engine.IO "message" type
- 0 for Socket.IO "connect" type
- and then the JSON-encoded payload
```

Please note that Postman now has first-class support for Socket.IO: https://blog.postman.com/postman-now-supports-socket-io/

**Reply by paramsiddharth** [reply]
Thank you for your answer!

![image](https://user-images.githubusercontent.com/30315706/131956515-726fcc1f-1eb1-47e1-b747-13f2624c2a59.png)

Have a look at the screenshot. Can you please guide me where I should put the `auth` object, the query parameters, the headers, or the payload, before clicking "Connect"?

**Reply by darrachequesne** [reply]
Oh, I see, it seems the `auth` object is not supported yet. You can use the raw websocket then:

![Screenshot from 2021-09-03 08-19-16](https://user-images.githubusercontent.com/13031701/131959522-9848a0a4-df31-4d08-830e-f0169fc73558.png)

**Reply by ohhjee** [reply]
Facing same issue

**Comment by imanfattahi** [comment]
Hi
I don't know why Postman doesn't support this important feature. You can easily use the **Firecamp** Google extension instead.

**Reply by paramsiddharth** [reply]
Thank you for the answer!

**Reply by theguywhocodes96** [reply]
@imanfattahi I am facing an issue on Firecamp while connecting to my local socket.io connection. Getting the following errors:
![image](https://github.com/user-attachments/assets/b253a213-c02e-4ae6-8586-b0f93596b83c)

Tried passing both localhost and with the IP address in the URL.
![image](https://github.com/user-attachments/assets/8a1674f2-9ff0-42b9-badf-14db8745ca03)

Any suggestions on how to resolve it?
